### PR TITLE
Add '--config' flag to specify a configuration file to the build command

### PR DESCRIPTION
**Pull request by NatashaWso2**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/14860

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by pubudu91** [comment]
We didn't agree upon anything for this right? Like would this be misleading (considering the config file here is a runtime thing required for the tests)?

**Comment by NatashaWso2** [comment]
@pubudu91 added the config flag here since there is an inconsistency where we allow the '--config' flag in the test command but not in the build command.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java [hunk 1]
```diff
@@ -182,6 +185,10 @@ public void execute() {
                 BuilderUtils.compileAndWriteJar(sourceRootPath, pkgName, targetFileName, buildCompiledPkg,
                         offline, lockEnabled, skiptests, experimentalFlag, dumpBIR);
             } else {
+                // Load the configuration file. If no config file is given then the default config file i.e.
+                // "ballerina.conf" in the source root path is taken.
+                LauncherUtils.loadConfigurations(sourceRootPath, configFilePath);
```
Comment (pubudu91) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: Doesn't this mean that we won't be able to give a config file to the build command if the target is the JVM?

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/util/TesterinaUtils.java [hunk 2]
```diff
@@ -198,10 +196,7 @@ public static void setManifestConfigs(Path sourceRoot) {
      */
     public static void executeTests(Path sourceRootPath, Map<BLangPackage, CompiledBinaryFile.ProgramFile>
             programFileMap) {
-        // Load configuration file. The default config file is taken "ballerina.conf" in the source root path
-        LauncherUtils.loadConfigurations(sourceRootPath, new HashMap<>(), null, false);
```
Comment (pubudu91) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/util/TesterinaUtils.java]: Why was this removed? Wouldn't this cause problems if the user runs `ballerina test` with a config file provided?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java (hunk 1 above)
Comment (NatashaWso2) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/cmd/BuildCommand.java]: We should allow it. Fixed with 0071468. Thanks :)
